src/components/WhatWeb/WhatWeb.tsx

```diff
@@ -0,0 +1,198 @@
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            // If the process was terminated successfully, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+                // If the process was terminated due to a signal, display the signal code.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+                // If the process was terminated with an error, display the exit code and signal code.
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+
+            // Now that loading has completed, allow the user to save the output to a file.
+            setAllowSave(true);
+            setHasSaved(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    // Actions taken after saving the output
+    const handleSaveComplete = () => {
+        // Indicating that the file has saved which is passed
+        // back into SaveOutputToTextFile to inform the user
+        setHasSaved(true);
+        setAllowSave(false);
+    };
+
+    const onSubmit = async (values: FormValuesType) => {
+        // Disallow saving until the tool's execution is complete
+        setAllowSave(false);
+
+        // Activate loading state to indicate ongoing process
+        setLoading(true);
+
+        // Construct arguments for the aircrack-ng command based on form input
+        const args = [values.url];
+
+        // Execute the whatweb command via helper method and handle its output or potential errors
+        CommandHelper.runCommandGetPidAndOutput("whatweb", args, handleProcessData, handleProcessTermination)
+
+            .then(({ pid, output }) => {
+                // Update the output with the results of the command execution.
+                setOutput(output);
+
+                // Store the process ID of the executed command.
+                setPid(pid);
+                console.log("The then statement has been run");
+            })
+            .catch((error) => {
+                // Display any errors encountered during command execution.
+                setOutput(error.message);
+                console.log("An error has been caught");
+                // Deactivate loading state.
+                setLoading(false);
+            });
+        console.log("This is the PID: " + pid);
+        console.log("args is " + args);
```

Comment:
- [ ] remove console logs for non errors

Reply:
as a side note there is many libraries that we should be using and I dont believe there is one currently installed in the project.